app/components/SearchResults.tsx

```diff
@@ -0,0 +1,90 @@
+import { Link } from "@remix-run/react"
+import React from "react"
+import type JSXNode from "~/types/JSXNode"
+import SanitizedTextSpan from "./SanitizedTextSpan"
+
+interface SearchResultsProps {
+  data: any
+  text: string
+  pageNumber: string | undefined
+}
+
+function displayEntries(pageNumber: string, text: string, data: any) {
+  if (data.entries === undefined || data.entries.length === 0) {
+    return null
+  }
+
+  return (
+    <>
+      <h3 className="text-xl font-bold">
+        <>
+          Entries containing &ldquo;{text}&rdquo;: {data.entries.length}
+        </>
+      </h3>
+      {data.entries.map((e: any) => {
```

Comment:
Yeah also same thing as the other comment-- unfortunately even with the prop refactored to `Record<string, any[]> I still have the any[] array as a type (which yeah that's my bad-- I for some reason didn't think inheritance existed, I think, so I did it that way so I could get all different types of paginated results. This is another thing that should change in the next PR I think, since we're rewriting the backend basically entirely.)

Though I did make minor changes! So it's a bit better in the meantime. So the `:any` is indeed gone.

(I'll start the next PR immediately btw-- it'll be the first thing I push.)